### PR TITLE
Add `background` parameter to `overline`, `underline` and `strike`

### LINKED ISSUE #1771: Add foreground/background option to underline
### Description

If you make a really thick underline, currently you get this:
image
However, sometimes you would rather have the underline behind the content.

Hence, this feature request asks for the option to have the underline in the foreground/background as part of the `underline` function.

### Use Case

This can be useful when creating stylized content in graphic design.

### PR DESCRIPTION
**Pull request by tingerrr**
Adds a `background` parameter to `overline` and `underline`, resolves #1771.

### CODE REVIEW

**Comment by laurmaedje** [comment]
Should `strike` also support this?

File: crates/typst-library/src/text/deco.rs [hunk 1]
```diff
@@ -67,7 +67,6 @@ pub struct UnderlineElem {
     /// #underline(background: true)[This is stylized.] \
     /// #underline(background: false)[This is partially hidden.]
     /// ```
-    #[default(false)]
```
Comment (laurmaedje) [inline crates/typst-library/src/text/deco.rs]: This removal is probably influenced by the quotes PR, but for boolean I actually prefer to have them explicit. The implicit false is not as obvious as an implicit auto or none.

Comment (tingerrr) [reply crates/typst-library/src/text/deco.rs]: Sure thing!

**Comment by laurmaedje** [comment]
Thank you!

**Pull request merged**